WdtBase.cpp

```diff
@@ -163,6 +171,24 @@ WdtTransferRequest::WdtTransferRequest(const string& uriString) {
       }
     }
   } while (!portsList.empty());
+  string startPortStr = wdtUri.getQueryParam(START_PORT_PARAM);
+  string numPortsStr = wdtUri.getQueryParam(NUM_PORTS_PARAM);
+  if (!startPortStr.empty() && !numPortsStr.empty()) {
```

Comment:
I should be able to specify only 1 of those - ie || not &&
also we should get the start_port from  host:port part of url (too, even if it's good to support query args with same name as flag as well)